pokemon-api/src/main/java/cz/muni/fi/pa165/pokemon/league/participation/manager/dto/BadgeStatusChangeDTO.java

```diff
@@ -0,0 +1,72 @@
+package cz.muni.fi.pa165.pokemon.league.participation.manager.dto;
+
+import com.sun.istack.internal.NotNull;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.enums.ChallengeStatus;
+
+import java.util.Objects;
+
+/**
+ * DTO for change of badge status
+ *
+ * @author Michal Mokros 456442
+ */
+public class BadgeStatusChangeDTO {
+
+    @NotNull
+    private Long badgeId;
+
+    @NotNull
+    private Long trainerId;
+
+    private ChallengeStatus newStatus;
```

Comment:
This is expressed in the semantics of the called method. If you want to use a private helper method that actually does the state changes, pass the newStatus to it as a another argument.

Reply:
Therefore remove it?